lib/CodeBuilder/Adapter/WorseReflection/WorseBuilderFactory.php

```diff
@@ -38,20 +39,13 @@ public function fromSource($source): SourceCodeBuilder
         $classes = $this->reflector->reflectClassesIn($source);
         $builder = SourceCodeBuilder::create();
 
-        foreach ($classes as $class) {
-            if ($class->isClass()) {
-                $this->build('class', $builder, $class);
-                continue;
-            }
-
-            if ($class->isInterface()) {
-                $this->build('interface', $builder, $class);
-                continue;
-            }
-
-            if ($class->isTrait()) {
-                $this->build('trait', $builder, $class);
-                continue;
+        foreach ($classes as $classLike) {
+            if ($classLike instanceof ReflectionClass) {
+                $this->build('class', $builder, $classLike);
+            } elseif ($classLike instanceof ReflectionInterface) {
+                $this->build('interface', $builder, $classLike);
+            } elseif ($classLike instanceof ReflectionTrait) {
+                $this->build('trait', $builder, $classLike);
             }
```

Comment:
i'd rather avoid `else` in the code base. can we have continue instead?

Reply:
Yeah, switch would have been the best but I was unsure if it would be able to handle `instanceof`.

Reply:
```
if ($cond) {
    // action;
    continue;
}
```

is fine